### ISSUE TITLE
snapshot_schedule module is not idempotent

### ISSUE THREAD

**Issue by justinc1** [ScaleComputing/HyperCoreAnsibleCollection#18]
Reported by @ddemlow 

```
I added the following task … the schedule gets created and appears fine, but shows changed every time I run the play book.  ideas?

  - name: Setup snapshot schedule
    scale_computing.hypercore.snapshot_schedule:
      name: snap-daily-midnight
      state: present
      recurrences:
        - name: daily-midnight
          frequency: "FREQ=DAILY;INTERVAL=1"  # RFC-2445
          start: "2010-01-01 00:00:00"
          local_retention: "{{ 7*24*60*60 }}"  # 7 days, unit seconds
          remote_retention: "{{ 1*24*60*60 }}" # optional, None or 0 means same as local_retention.

-vvv

TASK [Setup snapshot schedule] ***********************************************************************************************************************
task path: /Users/davedemlow/ansible_collections/Edge_Demo_Playbook.yml:107
changed: [192.168.1.242] => changed=true 
  diff:
    after:
      name: snap-daily-midnight
      recurrences:
      - frequency: FREQ=DAILY;INTERVAL=1
        local_retention: 604800
        name: daily-midnight
        remote_retention: 86400
        replication: true
        start: '2010-01-01 00:00:00'
        uuid: df188982-627f-4a06-a671-64c9d4ca974b
      uuid: 82a9995d-7d8d-4161-8063-241934f2bccc
    before:
      name: snap-daily-midnight
      recurrences:
      - frequency: FREQ=DAILY;INTERVAL=1
        local_retention: 604800
        name: daily-midnight
        remote_retention: 86400
        replication: true
        start: '2010-01-01 00:00:00'
        uuid: 9dad1c64-57ac-4900-b124-1aecacb72dfd
      uuid: 82a9995d-7d8d-4161-8063-241934f2bccc
  invocation:
    module_args:
      cluster_instance:
        host: https://192.168.1.242/
        password: VALUE_SPECIFIED_IN_NO_LOG_PARAMETER
        timeout: null
        username: VALUE_SPECIFIED_IN_NO_LOG_PARAMETER
      name: snap-daily-midnight
      recurrences:
      - frequency: FREQ=DAILY;INTERVAL=1
        local_retention: 604800
        name: daily-midnight
        remote_retention: 86400
        start: '2010-01-01 00:00:00'
      state: present
  record:
  - name: snap-daily-midnight
    recurrences:
    - frequency: FREQ=DAILY;INTERVAL=1
      local_retention: 604800
      name: daily-midnight
      remote_retention: 86400
      replication: true
      start: '2010-01-01 00:00:00'
      uuid: df188982-627f-4a06-a671-64c9d4ca974b
    uuid: 82a9995d-7d8d-4161-8063-241934f2bccc
```

Seems like integration test for snapshot_schedule lacks idempotence test. We need to to test 2 cases:
- whole snapshot schedule is not present, module creates it on first invocation, module does not modify snapshot schedule a second time
- whole snapshot schedule is present, module adds (or removes) a recurrence rule on first invocation, module does not modify snapshot schedule a second time

**Comment by tjazsch** [comment]
Thank you for reporting the bug. Indeed, when running again, the result always showed `changed=True` with no changed actually made. In process of resolving.

**Comment by tjazsch** [comment]
Resolved by #34
